Source/WebInspectorUI/UserInterface/Views/RecordingTraceDetailsSidebarPanel.js

```diff
@@ -75,19 +75,18 @@ WI.RecordingTraceDetailsSidebarPanel = class RecordingTraceDetailsSidebarPanel e
         if (!this._action)
             return;
 
-        let trace = this._action.trace;
-        this._backtraceTreeController.callFrames = trace;
-
-        if (!trace.length) {
-            let noTraceDataElement = this.contentView.element.appendChild(document.createElement("div"));
-            noTraceDataElement.classList.add("no-trace-data");
-
-            let noTraceDataMessageElement = noTraceDataElement.appendChild(document.createElement("div"));
-            noTraceDataMessageElement.classList.add("message");
-            noTraceDataMessageElement.textContent = WI.UIString("Call Stack Unavailable");
+        let stackTrace = this._action.stackTrace;
+        if (!stackTrace?.callFrames.length) {
+            let noStackTraceContainerElement = this.contentView.element.appendChild(document.createElement("div"));
+            noStackTraceContainerElement.classList.add("no-stack-trace");
+
+            let noStackTraceMessageElement = noStackTraceContainerElement.appendChild(document.createElement("div"));
+            noStackTraceMessageElement.classList.add("message");
+            noStackTraceMessageElement.textContent = WI.UIString("Call Stack Unavailable");
             return;
         }
 
+        this._backtraceTreeController.stackTrace = stackTrace;
```

Comment:
Is this change in behavior okay? This used to get set even if the stack trace was empty.

Reply:
oh we probably should still do this anyways just so that `_stackTrace` is set, but yeah it was basically a no-op before and is still one after